src/lib/operators/table_scan.cpp

```diff
@@ -174,16 +175,20 @@ void TableScan::_init_scan() {
     return;
   }
 
+  if (_scan_type == ScanType::OpIsNull || _scan_type == ScanType::OpIsNotNull) {
+    _impl = std::make_unique<IsNullTableScanImpl>(_in_table, _left_column_id, _scan_type);
+    return;
+  }
+
   if (is_variant(_right_parameter)) {
     const auto right_value = boost::get<AllTypeVariant>(_right_parameter);
 
     if (variant_is_null(right_value)) {
-      _impl = std::make_unique<IsNullTableScanImpl>(_in_table, _left_column_id, _scan_type);
-
-      return;
+      // early out because of NULL semantics
```

Comment:
Please extend on this. Maybe something like "because nothing can be `= NULL`, only `IS NULL`".